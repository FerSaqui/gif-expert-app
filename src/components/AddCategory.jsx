import React, { useState } from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ({ onNewCategory }) => {
    const [inputValue, setInputValue] = useState("");

    const onInputChange = ({ target }) => {
        setInputValue( target.value );
    }

    const handleSubmit = ( event ) => {
        event.preventDefault();

        if ( inputValue.trim().length <= 1 ) return;

        //Props que recibe este componente del componente padre (GifExpertApp), es una función
        //onAddCategory( categorias => [inputValue, ...categorias] );

        //Se emite este método
        onNewCategory( inputValue );
        setInputValue("");
    }

  return (
    <form onSubmit={ handleSubmit } aria-label="form">
        <input 
            type={ "text" }
            placeholder= "Buscar GIFs"
            value={ inputValue }
            onChange= { onInputChange }
        />
    </form>

  )
}

AddCategory.propTypes = {
    onNewCategory: PropTypes.func.isRequired
}