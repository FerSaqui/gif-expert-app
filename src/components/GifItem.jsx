import PropTypes from 'prop-types';

export const GifItem = ({ title, url }) => {
  return (
    <div className="container-card">
        <img className="container-card__img" src={ url } alt={ title }/>
        <p className="container-card__text">{ title }</p>
    </div>
  )
}

GifItem.propTypes = {
  title:  PropTypes.string.isRequired,
  url:    PropTypes.string.isRequired
}