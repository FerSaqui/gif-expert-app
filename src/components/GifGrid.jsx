//import { useEffect, useState } from "react";
//import { getGifs } from "../helpers/getGifs";
import { useFetchGifs } from "../hooks/useFetchGifs";
import { GifItem } from "./GifItem";

import PropTypes from 'prop-types';

export const GifGrid = ({ category }) => {
  const { newImages, isLoading } = useFetchGifs( category );

  return (
    <>
        <h2 className="category-text">{ category }</h2>
        {
          isLoading && (<h2>Cargando...</h2>)
        }
        <div className="container">
          {
            /**Se puso parentesis porque es un return implÃ­cito (que no se coloca la palabra return) */
            newImages.map( ( img ) => (
              <GifItem 
                key={ img.id }
                { ...img }
              />
            ))
          }
        </div>
    </>
  )
}

GifGrid.propTypes = {
  category: PropTypes.string.isRequired
}