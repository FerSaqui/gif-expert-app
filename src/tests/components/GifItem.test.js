import { render, screen } from "@testing-library/react";
import { GifItem } from "../../components/GifItem";

describe('Pruebas en <GifItem />', () => {
    const title = "Hola mundo";
    const url = "https://";

    test('Debe de hacer match con el snapshot', () => {
        const { container } = render( <GifItem title={ title } url={ url } /> );
        expect( container ).toMatchSnapshot();
    });

    test('Debe de mostrar la imágen con el URL y el ALT indicado', () => {
        render( <GifItem title={ title } url={ url } /> );

        const { src, alt } = screen.getByRole("img");
        expect( src ).toBe( url );
        expect( alt ).toBe( title );
    });

    test('Debe de mostrar el título en el componente', () => {
        render( <GifItem title={ title } url={ url } /> );

        expect( screen.getByText( title ) ).toBeTruthy();
    });

});