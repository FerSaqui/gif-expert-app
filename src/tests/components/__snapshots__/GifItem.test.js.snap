// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifItem /> Debe de hacer match con el snapshot 1`] = `
<div>
  <div
    class="container-card"
  >
    <img
      alt="Hola mundo"
      class="container-card__img"
      src="https://"
    />
    <p
      class="container-card__text"
    >
      Hola mundo
    </p>
  </div>
</div>
`;
